plugins/rhacm-backend/src/providers/ManagedClusterProvider.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-node';
+import {
+  ANNOTATION_ORIGIN_LOCATION,
+  ANNOTATION_LOCATION,
+} from '@backstage/catalog-model'
+import { CustomObjectsApi } from "@kubernetes/client-node"
+import { getManagedCluster, getManagedClusters, hubApiClient } from '../helpers/kubernetes';
+import { CONSOLE_CLAIM, HUB_CLUSTER_NAME_IN_ACM } from '../constants';
+import { getClaim } from '../helpers/parser';
+import { getHubClusterName } from '../helpers/config';
+
+/**
+ * Provides entities from AWS EKS Cluster service.
+ */
+export class ManagedClusterProvider implements EntityProvider {
+  protected readonly client: CustomObjectsApi;
+  protected readonly hubName: string;
+  protected readonly logger: winston.Logger;
+  protected connection?: EntityProviderConnection;
+
+  protected constructor(client: CustomObjectsApi, hubName: string, options: { logger: winston.Logger }) {
+    this.client = client;
+    this.hubName = hubName;
+    this.logger = options.logger;
+  }
+
+  static fromConfig(config: Config, options: { logger: winston.Logger }) {
+    const client = hubApiClient(config, options.logger);
+    const hubName = getHubClusterName(config);
+
+    return new ManagedClusterProvider(client, hubName, options);
+  }
+  public async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+  }
+
+  getProviderName(): string {
+    return 'rhacm-managed-cluster';
+  }
+
+  async run(): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    this.logger.info(
+      `Providing OpenShift cluster resources from RHACM`,
+    );
+    const hubConsole = getClaim(await getManagedCluster(this.client, HUB_CLUSTER_NAME_IN_ACM), CONSOLE_CLAIM)
+
+    const resources: ResourceEntity[] = (await getManagedClusters(this.client) as {items: Array<any>}).items.map((i) => {
+      return {
+        kind: 'Resource',
+        apiVersion: 'backstage.io/v1beta1',
+        metadata: {
+          name: i.metadata.name,
+          title: i.metadata?.labels?.name,
+          annotations: {
+            [ANNOTATION_LOCATION]: `${this.getProviderName()}:${this.hubName}`,
+            [ANNOTATION_ORIGIN_LOCATION]: `${this.getProviderName()}:${this.hubName}`,
+          },
+          links: [
+            {
+              url: getClaim(i, CONSOLE_CLAIM),
+              title: "OpenShift Console",
+              icon: "dashboard"
+            },
+            {
+              url: `${hubConsole}/multicloud/infrastructure/clusters/details/${i.metadata.name}/`,
+              title: "RHACM Console",
+            },
+            i.metadata?.labels?.clusterID && {
+              url: `https://console.redhat.com/openshift/details/s/${i.metadata.labels.clusterID}`,
+              title: 'OpenShift Cluster Manager',
+            }
+          ]
+        },
+        spec: {
+          owner: 'unknown',
```

Comment:
Is it possible to use a configuration value in the config file for example:
```yaml
rhacm:
    hubOwner: 'group:operate-first'
```
and use it here instead of `unknown` owner?

Reply:
I don't know... do we want to assume there's always a single owner to all the clusters? In ACM there's a concept of cluster ownership which gets under some conditions applied.. but I don't understand it yet. 

https://github.com/stolostron/console/blob/24de3bb55e229d393334474bb365040eb048ea03/frontend/src/resources/utils/get-cluster.ts#L291

I don't know.. we should investigate it further as a separate issue, wdyt?

Reply:
Yea sure, that make sense.